<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MicroPythonPage</class>
 <widget class="QWidget" name="MicroPythonPage">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>476</width>
    <height>869</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QLabel" name="headerLabel">
     <property name="text">
      <string>&lt;b&gt;Configure MicroPython&lt;/b&gt;</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="Line" name="line9_3">
     <property name="frameShape">
      <enum>QFrame::HLine</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Sunken</enum>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>Serial Link</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="0">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Timeout for Serial Link Communication:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QSpinBox" name="timeoutSpinBox">
        <property name="toolTip">
         <string>Enter the timout value</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
        </property>
        <property name="suffix">
         <string> s</string>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>30</number>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>195</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="1" column="0" colspan="3">
       <widget class="QCheckBox" name="syncTimeCheckBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>Select to synchronize the time after connection is established</string>
        </property>
        <property name="text">
         <string>Synchronize Time at Connect</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>REPL Pane</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Color Scheme for ANSI Escape Codes:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QComboBox" name="colorSchemeComboBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>Select the color scheme to be applied for ANSI color escape codes</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0" colspan="2">
       <widget class="QCheckBox" name="replWrapCheckBox">
        <property name="toolTip">
         <string>Select to wrap long line in the REPL pane</string>
        </property>
        <property name="text">
         <string>Wrap long lines</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_3">
     <property name="title">
      <string>MPY Cross Compiler</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Program:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="E5PathPicker" name="mpyCrossPicker" native="true">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="focusPolicy">
         <enum>Qt::WheelFocus</enum>
        </property>
        <property name="toolTip">
         <string>Enter the path of the cross compiler executable</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_5">
     <property name="title">
      <string>PyBoard</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="QLabel" name="label_7">
        <property name="text">
         <string>dfu-util Path:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="E5PathPicker" name="dfuUtilPathPicker" native="true">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="focusPolicy">
         <enum>Qt::WheelFocus</enum>
        </property>
        <property name="toolTip">
         <string>Enter the path of the dfu-util flashing executable</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_6">
     <property name="title">
      <string>Firmware</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_4">
      <item row="0" column="0">
       <widget class="QLabel" name="label_8">
        <property name="text">
         <string>MicroPython:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="E5ClearableLineEdit" name="micropythonFirmwareUrlLineEdit">
        <property name="toolTip">
         <string>Enter the URL for the MicroPython firmware for PyBoard, ESP8266 and ESP32</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_10">
        <property name="text">
         <string>CircuitPython:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="E5ClearableLineEdit" name="circuitpythonFirmwareUrlLineEdit">
        <property name="toolTip">
         <string>Enter the URL for the CircuitPython firmware</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_9">
        <property name="text">
         <string>BBC micro:bit:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="E5ClearableLineEdit" name="microbitFirmwareUrlLineEdit">
        <property name="toolTip">
         <string>Enter the URL for the BBC micro:bit Firmware</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_4">
     <property name="title">
      <string>Documentation</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_3">
      <item row="0" column="0">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>MicroPython:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="E5ClearableLineEdit" name="micropythonDocuUrlLineEdit">
        <property name="toolTip">
         <string>Enter the URL for the MicroPython documentation</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>CircuitPython:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="E5ClearableLineEdit" name="circuitpythonDocuUrlLineEdit">
        <property name="toolTip">
         <string>Enter the URL for the CircuitPython documentation</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>BBC micro:bit:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="E5ClearableLineEdit" name="microbitDocuUrlLineEdit">
        <property name="toolTip">
         <string>Enter the URL for the BBC micro:bit MicroPython documentation</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>252</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>E5PathPicker</class>
   <extends>QWidget</extends>
   <header>E5Gui/E5PathPicker.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>E5ClearableLineEdit</class>
   <extends>QLineEdit</extends>
   <header>E5Gui/E5LineEdit.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>timeoutSpinBox</tabstop>
  <tabstop>syncTimeCheckBox</tabstop>
  <tabstop>colorSchemeComboBox</tabstop>
  <tabstop>replWrapCheckBox</tabstop>
  <tabstop>mpyCrossPicker</tabstop>
  <tabstop>dfuUtilPathPicker</tabstop>
  <tabstop>micropythonFirmwareUrlLineEdit</tabstop>
  <tabstop>circuitpythonFirmwareUrlLineEdit</tabstop>
  <tabstop>microbitFirmwareUrlLineEdit</tabstop>
  <tabstop>micropythonDocuUrlLineEdit</tabstop>
  <tabstop>circuitpythonDocuUrlLineEdit</tabstop>
  <tabstop>microbitDocuUrlLineEdit</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
